### PR TITLE
Refs [TASK][36602] Data saved screen

### PR DESCRIPTION
**Pull request by manhhung23399**
## Related Tickets
- [#Task 36602: Data saved screen](https://edu-redmine.sun-asterisk.vn/issues/36602)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/gamevui/ui/SavedFragment.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import com.sun.gamevui.R
+import com.sun.gamevui.base.BaseFragment
+import com.sun.gamevui.data.model.Game
+import com.sun.gamevui.databinding.FragmentSavedBinding
+import com.sun.gamevui.ui.adapter.SavedGamesAdapter
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class SavedFragment : BaseFragment<FragmentSavedBinding>() {
+
+    private val savedGamesAdapter = SavedGamesAdapter(this::onItemClick)
```
Comment (trangnq-2533) [inline app/src/main/java/com/sun/gamevui/ui/SavedFragment.kt]:
```suggestion
    private val savedGamesAdapter = SavedGamesAdapter(::onItemClick)
```

File: app/src/main/java/com/sun/gamevui/ui/adapter/SavedGamesAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import com.sun.gamevui.R
+import com.sun.gamevui.base.BaseAdapter
+import com.sun.gamevui.base.BaseViewHolder
+import com.sun.gamevui.data.model.Game
+import com.sun.gamevui.databinding.ItemGameSavedBinding
+
+
```
Comment (trangnq-2533) [inline app/src/main/java/com/sun/gamevui/ui/adapter/SavedGamesAdapter.kt]: hai dòng trống :v

Comment (manhhung23399) [reply app/src/main/java/com/sun/gamevui/ui/adapter/SavedGamesAdapter.kt]: dở thật, hehe

File: app/src/main/java/com/sun/gamevui/data/GameLocalDataSource.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+class GameLocalDataSource(
+    private val gameDao: GameDao
+) : GameDataSource.Local {
+    override suspend fun getSavedGames() =
+        gameDao.getSavedGames()
+
+    override suspend fun insertGame(game: Game) {
+        gameDao.insertGame(game)
+    }
```
Comment (tuannm-1276) [inline app/src/main/java/com/sun/gamevui/data/GameLocalDataSource.kt]:
```suggestion
    override suspend fun insertGame(game: Game) = gameDao.insertGame(game)
 
```

File: app/src/main/java/com/sun/gamevui/data/GameLocalDataSource.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    override suspend fun deleteGame(game: Game) {
+        gameDao.deleteGame(game)
+    }
```
Comment (tuannm-1276) [inline app/src/main/java/com/sun/gamevui/data/GameLocalDataSource.kt]:
```suggestion
    override suspend fun deleteGame(game: Game) = gameDao.deleteGame(game)
    
```

**Pull request merged**